### PR TITLE
Update PPP and fix PE-D bugs

### LINKED ISSUE #111: [PE-D][Tester E] GUI change not reflected for setlesson
It was stated in the UG that when the students number of lessons exceeded the number of lessons it is automatically reduced to number of lessons. However when adjusting lessons to 5, the changes were not reflected as seen in the image. While the student attendance was changed, the total number of lesson was not changed. the Intended output should be 5/5 as seen in the UG. 

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: tayruxin/ped#7

### PR DESCRIPTION
**Pull request by seanpzk**
Updates:
- PPP
- Added **Planned Enhancements** in DG
- Added test cases for `SetLessonCommandParser`, `StartLessonCommandParser`, `RemoveStudentCommandParser`
- Removed `remove` command to standardise with other commands

Issues fixed:
- #125 
- #96 
- #84 
- #135
- #121 

Issues to be handled under Planned Enhancements:
- #134 
- #133 
- #120 
- #109 
- #107

### CODE REVIEW

**Review by songgthu — APPROVED** [review]
LGTM!

File: docs/UserGuide.md [hunk 1]
```diff
@@ -627,9 +627,8 @@ This can be useful when you [start a lesson](#starting-a-lesson-startlesson-c) b
 <box type="info">
 
 **Information:**
-- NUMBER_OF_LESSONS takes numbers such as `0`,`1`,`2`,`3`
+- NUMBER_OF_LESSONS must be a valid non-negative integer such as `0`,`1`,`2`,`3`
```
Comment (DonovanJJ) [inline docs/UserGuide.md:630]: Just nitpicking... perhaps you could say "positive" integer instead

File: docs/UserGuide.md [hunk 2]
```diff
@@ -872,7 +871,7 @@ There is no need to save manually.
...
 |-----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| **add**         | **Adding a class:** <br> **Format:** `add /c CLASS_NAME` <br> **E.g:** `add /c CS2103T` <br> <br> **Adding a student:** <br> **Format:** `add /s STUDENT_NAME  /c CLASS_NAME` <br> **E.g:** `add /s John /c CS2103T` <br><br> **Adding a list of students:** <br> Coming soon                                                                            |
+| **add**         | **Adding a class:** <br> **Format:** `add /c CLASS_NAME` <br> **E.g:** `add /c CS2103T` <br> <br> **Adding a student:** <br> **Format:** `add /s STUDENT_NAME /c CLASS_INDEX` <br> **E.g:** `add /s John /c CS2103T` <br><br> **Adding a list of students:** <br> Coming soon                                                                            |
```
Comment (DonovanJJ) [inline docs/UserGuide.md:874]: I don't think we will be able to add a list of students to a class anymore due to the feature freeze... Perhaps removing the "Adding a list of students" might result in a more concise command summary for the users

File: src/main/java/seedu/address/logic/commands/RemoveStudentCommand.java [hunk 3]
```diff
@@ -18,18 +18,18 @@
 /**
  * Removes a Student identified by Name from a Class.
```
Comment (DonovanJJ) [inline src/main/java/seedu/address/logic/commands/RemoveStudentCommand.java:19]: Perhaps this could be updated to "Identified by Index" as well

**Review by DonovanJJ — APPROVED** [review]
LGTM! Just some nitpicking 👍

File: docs/UserGuide.md (hunk 1 above)
Comment (seanpzk) [reply docs/UserGuide.md:630]: Mine can accept 0 ahahah

Comment (seanpzk) [reply docs/UserGuide.md:630]: Mine can accept 0, that's why I did not follow the rest, thanks though!

File: docs/UserGuide.md (hunk 2 above)
Comment (seanpzk) [reply docs/UserGuide.md:874]: Okay I will change in future iteration for UG!

**Pull request merged**